Program.cs

```diff
@@ -1,10 +1,69 @@
-﻿namespace WeatherMonitoringSystem
+﻿using WeatherMonitoringSystem.model;
+using WeatherMonitoringSystem.Observer;
+using WeatherMonitoringSystem.Strategy;
+using WeatherMonitoringSystem.utils;
+
+namespace WeatherMonitoringSystem
 {
     internal class Program
     {
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello, World!");
+            Console.WriteLine("Weather Monitoring System");
+
+            try
+            {
+                var bots = FileConfig();
+               
+                WeatherSubject subject = new WeatherSubject();
+
+                AddBotObserversTOSubject(subject, bots);
+
+                WeatherData weather = UserInput();
+
+                subject.SetWeatherValue(weather);
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
         }
+
+        public static IEnumerable<Bot> FileConfig()
+        {
+            ConfigurationsFileReader configFileReader = new ConfigurationsFileReader();
```

Comment:
You can use `ConfigurationBuilder` from Microsoft. That way you'd be able to parse configurations as objects directly if you included the `Binder` package, and you can read JSON configuration easily using `Json` package (both packages are part of `ConfigurationBuilder` package).

It'd also provide strongly typed methods for getting the configurations. So you don't need to handle any parsing details.

```csharp
var rainBot = configuration.GetSection(nameof(RainBot)).Get<RainBot>();
```

https://www.daveabrock.com/2021/01/19/config-top-level-programs/